string action
int16 material
string grip_mode
float32[] coord
float32[] coord_2
